### PR TITLE
FIX query filter regex crashes

### LINKED ISSUE #2164: CB crashes when several regexp in q parameter and ~= operator
When retireving the list of entities with several regexp as follows:

DEBUG >>>>>>>>>>>>>	 Data sent:     	>>>>>>>>>>>>> 
	> Comp: CB
	> Url: http://192.168.21.64:1026/v2/entities
	> Method: get
	> Headers: {'fiware-service': 'eeee', 'fiware-servicepath': '/uuu', 'Accept': 'application/json'}
	> Params sent: {'q': 'humidity~=100;temperature>10', 'limit': 3, 'type': 'Room5'}

The CB crashes.

The oposite way ({'q':temperature>10;'humidity~=100', 'limit': 3, 'type': 'Room5'}) it works fine

{
  "orion" : {
  "version" : "1.1.0_20160520020628",
  "uptime" : "0 d, 0 h, 2 m, 31 s",
  "git_hash" : "c96ceaf0437c989a21c388c5015fe4059f3bccff",
  "compile_time" : "Fri May 20 02:08:44 CEST 2016",
  "compiled_by" : "develenv",
  "compiled_in" : "ci-iot-deven-01"
}
}

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #2164. In addition, it includes the CNR line for #2064, that we missed to include in a past PR.

Please, pay special attention to memory management logic (i.e. `new`s and `delete`s) in order to be sure that this PR is not introducing any leakage.

@kzangeli 
@crbrox

### CODE REVIEW

File: src/lib/rest/StringFilter.cpp [hunk 1]
```diff
@@ -920,28 +926,20 @@ bool StringFilter::parse(const char* q, std::string* errorStringP)
...
+    StringFilterItem* item = new StringFilterItem();
 
-    if (item.parse(s, errorStringP) == true)
+    if (item->parse(s, errorStringP) == true)
     {
       filters.push_back(item);
     }
     else
     {
+      free(item);
```
Comment (kzangeli) [inline src/lib/rest/StringFilter.cpp]: Better to use `delete'.
With `free`, the destructor is not called, and we've seen some problems in the past mixing C/C++ in malloc/new/free/delete situations. Not that I can recall any details  ...

Comment (fgalan) [inline src/lib/rest/StringFilter.cpp]: You are right... I copied free() for the line below, but delete must be use. We alredy know that scaring things may happen when new/delete is mixxed with malloc/free :)

Comment (fgalan) [inline src/lib/rest/StringFilter.cpp]: Fixed in 4a77e29

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**